### PR TITLE
Android x Migration

### PR DESCRIPTION
**Pull request by rkodev**
Android x Migration

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowPreferenceManager.java  1
         

Clones removed
==============
+ opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowPreferenceManager.java  1
         

Clones removed
==============
+ opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowPreferenceManager.java  1
         

Clones removed
==============
+ opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowPreferenceManager.java  1
         

Clones removed
==============
+ opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowPreferenceManager.java  1
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowAsyncTask.java  1
         

Clones removed
==============
+ opensrp-app/src/test/java/org/smartregister/sync/ClientProcessorTest.java  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowAsyncTask.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-app/src/test/java/org/smartregister/shadows/ShadowAsyncTask.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5750664&bid=18825973)

File: opensrp-app/build.gradle [hunk 1]
```diff
@@ -8,8 +8,9 @@ buildscript {
     }
     dependencies {
         classpath "com.android.tools.build:gradle:$androidToolsBuildGradle"
-        classpath 'org.apache.commons:commons-lang3:3.3.2'
-        classpath 'io.fabric.tools:gradle:1.30.0'
+        classpath 'org.apache.commons:commons-lang3:3.9'
+        classpath 'org.kt3k.gradle.plugin:coveralls-gradle-plugin:2.8.3'
```
Comment (githengi) [inline opensrp-app/build.gradle:12]: Remove this, this is an older plugin that is currently being used

File: opensrp-app/build.gradle [hunk 2]
```diff
@@ -232,25 +244,24 @@ dependencies {
...
     }
-    testImplementation('org.robolectric:shadows-multidex:4.0') {
+
+    testImplementation('org.robolectric:shadows-multidex:4.3.1') {
         exclude group: 'com.google.guava', module: 'guava'
     }
 
-    testImplementation "org.robolectric:shadows-support-v4:3.3.2"
     // PowerMock
     def powerMockVersion = '2.0.4'
```
Comment (ndegwamartin) [inline opensrp-app/build.gradle:266]: We can bump this to 2.0.7

**Review by githengi — APPROVED** [review]

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**